### PR TITLE
Prae spit extinguishes fire

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Will extinguish all mobs hit by the projectile (including human, so double edged sword). Will also remove some fire stacks from the tile, or extinguish the turf if the fire is weak enough

## Why It's Good For The Game

Marines have more and more way of setting xenos on fire. 
Cas was introduced, then the vali, then the plasma pistol, then the latest mortar buff.
Meanwhile, xeno keeps having only two way of "contering" flamers, acid wells (terrible, cost so much plasma, so long to refill),
acid spray(usefull)
and of course hivelord, with jelly. 

Basicly, if you are against flamer, you NEED an hivelord, and sometimes you just can't have it with your comp.

This gives prae some utility against flames, nothing too crazy. 
This will also make prae as usefull to counter fire as spitter; 30 seconds between each spray is not enough to extinguish players, unlike spitter who can do it reliably. 

## Changelog
:cl:
balance: Prae spit extinguishes fire
/:cl:

### CODE REVIEW

**Comment by SplinterGP** [comment]
i mean you can just extinguish yourself off by pressing B

**Comment by David-Stormwell** [comment]
Thought this was already a thing, guess not.
It's fine.

**Comment by SplinterGP** [comment]
what about instead of instant extinguish you can use it to reduce the firestacks/make the fire end earlier by reducing its remaining burn time

**Comment by BraveMole** [comment]
> what about instead of instant extinguish you can use it to reduce the firestacks/make the fire end earlier by reducing its remaining burn time

It does that on the turf?
And doing that on the mob would make that feature barely usefull

**Comment by SplinterGP** [comment]
i mean you can make it reduce some noticeable amount, instant extinguish is uh, a bit too much for me

File: code/modules/projectiles/ammo_datums.dm [hunk 1]
```diff
@@ -1624,6 +1624,18 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 	var/turf/T = get_turf(M)
 	if(!T)
 		T = get_turf(P)
+	for (var/A in T)
```
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1627]: Undescriptive af name

File: code/modules/projectiles/ammo_datums.dm [hunk 2]
```diff
@@ -1634,16 +1646,39 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 	if(O.density && !(O.flags_atom & ON_BORDER))
 		T = get_turf(get_step(T, turn(P.dir, 180))) //If the object is dense and not a border object like barricades, we instead drop in the location just prior to the target
 
+	for (var/A in T)		
+		if(istype(A, /obj/flamer_fire))
```
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1650]: ditto

File: code/modules/projectiles/ammo_datums.dm [hunk 3]
```diff
@@ -1634,16 +1646,39 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
 
 /datum/ammo/xeno/acid/heavy/on_hit_turf(turf/T,obj/projectile/P)
 	if(!T)
 		T = get_turf(P)
-
 	if(isclosedturf(T))
 		T = get_turf(get_step(T, turn(P.dir, 180))) //If the turf is closed, we instead drop in the location just prior to the turf
-
+	for (var/A in T)		
+		if(istype(A, /obj/flamer_fire))
```
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1671]: ditto

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by Hydrinator** [comment]
Yeah i don't see why not. Fire has been getting a bit carried away recently.

File: code/modules/projectiles/ammo_datums.dm [hunk 4]
```diff
@@ -1624,6 +1624,18 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
+			var/obj/flamer_fire/FF = atom
+			if(FF.firelevel > 20)
+				FF.firelevel -= 20
+				FF.updateicon()
+				continue
+			qdel(atom)
+			continue
+		if(istype(atom, /mob/living))
+			var/mob/living/target = atom
+			target.ExtinguishMob()
```
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1638]: Dont use atom it triggers syntax highlighting

File: code/modules/projectiles/ammo_datums.dm [hunk 5]
```diff
@@ -1634,16 +1646,39 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 	if(O.density && !(O.flags_atom & ON_BORDER))
 		T = get_turf(get_step(T, turn(P.dir, 180))) //If the object is dense and not a border object like barricades, we instead drop in the location just prior to the target
 
+	for (var/atom in T)
+		if(istype(atom, /obj/flamer_fire))
+			var/obj/flamer_fire/FF = atom
```
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1651]: ditto

File: code/modules/projectiles/ammo_datums.dm [hunk 6]
```diff
@@ -1634,16 +1646,39 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
 /datum/ammo/xeno/acid/heavy/on_hit_turf(turf/T,obj/projectile/P)
 	if(!T)
 		T = get_turf(P)
-
 	if(isclosedturf(T))
 		T = get_turf(get_step(T, turn(P.dir, 180))) //If the turf is closed, we instead drop in the location just prior to the turf
-
+	for (var/atom in T)
+		if(istype(atom, /obj/flamer_fire))
+			var/obj/flamer_fire/FF = atom
```
Comment (TiviPlus) [inline code/modules/projectiles/ammo_datums.dm:1672]: ditto

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**